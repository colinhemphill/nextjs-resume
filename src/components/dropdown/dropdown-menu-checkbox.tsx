'use client';

import { CheckIcon } from '@heroicons/react/24/solid';
import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
import { type ComponentPropsWithoutRef, ComponentRef, forwardRef } from 'react';
import { cn } from 'src/helpers/utils';

const DropdownMenuCheckboxItem = forwardRef<
  ComponentRef<typeof DropdownMenuPrimitive.CheckboxItem>,
  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.CheckboxItem>
>(({ className, children, checked, ...properties }, reference) => (
  <DropdownMenuPrimitive.CheckboxItem
    ref={reference}
    className={cn(
      'focus:bg-accent-11 focus:text-accent-1 relative flex cursor-default items-center rounded-sm py-1.5 pr-2 pl-8 text-sm transition-colors outline-none select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50',
      className,
    )}
    checked={checked}
    {...properties}
  >
    <span className="absolute left-2 flex h-3.5 w-3.5 items-center justify-center">
      <DropdownMenuPrimitive.ItemIndicator>
        <CheckIcon className="h-4 w-4" />
      </DropdownMenuPrimitive.ItemIndicator>
    </span>
    {children}
  </DropdownMenuPrimitive.CheckboxItem>
));

DropdownMenuCheckboxItem.displayName =
  DropdownMenuPrimitive.CheckboxItem.displayName;

export { DropdownMenuCheckboxItem };
