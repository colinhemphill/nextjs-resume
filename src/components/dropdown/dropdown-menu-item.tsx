'use client';

import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
import { type ComponentPropsWithoutRef, ComponentRef, forwardRef } from 'react';
import { cn } from 'src/helpers/utils';

const DropdownMenuItem = forwardRef<
  ComponentRef<typeof DropdownMenuPrimitive.Item>,
  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Item> & {
    inset?: boolean;
  }
>(({ children, className, inset, ...properties }, reference) => (
  <DropdownMenuPrimitive.Item
    ref={reference}
    className={cn(
      'focus:bg-accent-11 focus:text-accent-1 relative flex cursor-default items-center gap-2 rounded-sm px-2 py-1.5 text-sm transition-colors outline-none select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0',
      inset && 'pl-8',
      className,
    )}
    {...properties}
  >
    {children}
  </DropdownMenuPrimitive.Item>
));

DropdownMenuItem.displayName = DropdownMenuPrimitive.Item.displayName;

export { DropdownMenuItem };
