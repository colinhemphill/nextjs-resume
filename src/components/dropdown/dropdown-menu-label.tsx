'use client';

import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
import { type ComponentPropsWithoutRef, ComponentRef, forwardRef } from 'react';
import { cn } from 'src/helpers/utils';

const DropdownMenuLabel = forwardRef<
  ComponentRef<typeof DropdownMenuPrimitive.Label>,
  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Label> & {
    inset?: boolean;
  }
>(({ children, className, inset, ...properties }, reference) => (
  <DropdownMenuPrimitive.Label
    ref={reference}
    className={cn(
      'text-neutral-11 px-2 py-1.5 text-sm font-semibold',
      inset && 'pl-8',
      className,
    )}
    {...properties}
  >
    {children}
  </DropdownMenuPrimitive.Label>
));

DropdownMenuLabel.displayName = DropdownMenuPrimitive.Label.displayName;

export { DropdownMenuLabel };
