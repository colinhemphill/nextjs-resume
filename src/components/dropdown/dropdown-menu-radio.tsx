'use client';

import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
import { type ComponentPropsWithoutRef, ComponentRef, forwardRef } from 'react';
import { cn } from 'src/helpers/utils';

const DropdownMenuRadioGroup = DropdownMenuPrimitive.RadioGroup;

const DropdownMenuRadioItem = forwardRef<
  ComponentRef<typeof DropdownMenuPrimitive.RadioItem>,
  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.RadioItem>
>(({ children, className, ...properties }, reference) => (
  <DropdownMenuPrimitive.RadioItem
    ref={reference}
    className={cn(
      'focus:bg-accent-11 focus:text-accent-1 relative flex cursor-default items-center rounded-sm py-1.5 pr-2 pl-8 text-sm transition-colors outline-none select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50',
      className,
    )}
    {...properties}
  >
    <span className="absolute left-2 flex h-3.5 w-3.5 items-center justify-center">
      <DropdownMenuPrimitive.ItemIndicator>
        <span className="block h-2 w-2 rounded-full bg-current" />
      </DropdownMenuPrimitive.ItemIndicator>
    </span>
    {children}
  </DropdownMenuPrimitive.RadioItem>
));

DropdownMenuRadioItem.displayName = DropdownMenuPrimitive.RadioItem.displayName;

export { DropdownMenuRadioGroup, DropdownMenuRadioItem };
