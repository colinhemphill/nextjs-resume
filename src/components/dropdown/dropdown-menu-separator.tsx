'use client';

import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
import { type ComponentPropsWithoutRef, ComponentRef, forwardRef } from 'react';
import { cn } from 'src/helpers/utils';

const DropdownMenuSeparator = forwardRef<
  ComponentRef<typeof DropdownMenuPrimitive.Separator>,
  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Separator>
>(({ children, className, ...properties }, reference) => (
  <DropdownMenuPrimitive.Separator
    ref={reference}
    className={cn('bg-neutral-6 -mx-1 my-1 h-px', className)}
    {...properties}
  >
    {children}
  </DropdownMenuPrimitive.Separator>
));

DropdownMenuSeparator.displayName = DropdownMenuPrimitive.Separator.displayName;

export { DropdownMenuSeparator };
