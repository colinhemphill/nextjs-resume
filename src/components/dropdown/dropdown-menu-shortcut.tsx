'use client';

import { ReactNode, type HTMLAttributes } from 'react';
import { cn } from 'src/helpers/utils';

function DropdownMenuShortcut({
  children,
  className,
  ...properties
}: HTMLAttributes<HTMLSpanElement>): ReactNode {
  return (
    <span
      className={cn('ml-auto text-xs tracking-widest opacity-60', className)}
      {...properties}
    >
      {children}
    </span>
  );
}

DropdownMenuShortcut.displayName = 'DropdownMenuShortcut';

export { DropdownMenuShortcut };
