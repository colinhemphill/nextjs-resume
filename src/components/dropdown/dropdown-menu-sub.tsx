'use client';

import { ChevronRightIcon } from '@heroicons/react/24/solid';
import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
import { type ComponentPropsWithoutRef, ComponentRef, forwardRef } from 'react';
import { cn } from 'src/helpers/utils';

const DropdownMenuSub = DropdownMenuPrimitive.Sub;

const DropdownMenuSubContent = forwardRef<
  ComponentRef<typeof DropdownMenuPrimitive.SubContent>,
  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubContent>
>(({ children, className, ...properties }, reference) => (
  <DropdownMenuPrimitive.SubContent
    ref={reference}
    className={cn(
      'bg-neutral-1 text-neutral-12 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 border-neutral-6 z-50 min-w-[8rem] overflow-hidden rounded-md border p-1 shadow-lg',
      className,
    )}
    {...properties}
  >
    {children}
  </DropdownMenuPrimitive.SubContent>
));

DropdownMenuSubContent.displayName =
  DropdownMenuPrimitive.SubContent.displayName;

const DropdownMenuSubTrigger = forwardRef<
  ComponentRef<typeof DropdownMenuPrimitive.SubTrigger>,
  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubTrigger> & {
    inset?: boolean;
  }
>(({ className, inset, children, ...properties }, reference) => (
  <DropdownMenuPrimitive.SubTrigger
    ref={reference}
    className={cn(
      'focus:bg-accent-11 focus:text-neutral-1 data-[state=open]:bg-accent-11 data-[state=open]:text-neutral-1 flex cursor-default items-center gap-2 rounded-sm px-2 py-1.5 text-sm outline-none select-none [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0',
      inset && 'pl-8',
      className,
    )}
    {...properties}
  >
    {children}
    <ChevronRightIcon className="ml-auto" />
  </DropdownMenuPrimitive.SubTrigger>
));

DropdownMenuSubTrigger.displayName =
  DropdownMenuPrimitive.SubTrigger.displayName;

export { DropdownMenuSub, DropdownMenuSubContent, DropdownMenuSubTrigger };
