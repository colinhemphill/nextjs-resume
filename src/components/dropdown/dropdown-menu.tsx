'use client';

import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
import { type ComponentPropsWithoutRef, ComponentRef, forwardRef } from 'react';
import { cn } from 'src/helpers/utils';

const DropdownMenu = DropdownMenuPrimitive.Root;
const DropdownMenuGroup = DropdownMenuPrimitive.Group;
const DropdownMenuPortal = DropdownMenuPrimitive.Portal;
const DropdownMenuTrigger = DropdownMenuPrimitive.Trigger;

const DropdownMenuContent = forwardRef<
  ComponentRef<typeof DropdownMenuPrimitive.Content>,
  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Content>
>(({ children, className, sideOffset = 4, ...properties }, reference) => (
  <DropdownMenuPrimitive.Portal>
    <DropdownMenuPrimitive.Content
      ref={reference}
      sideOffset={sideOffset}
      className={cn(
        'data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 border-neutral-7 bg-neutral-1 text-neutral-12 z-50 min-w-[8rem] overflow-hidden rounded-md border p-1 shadow-md',
        className,
      )}
      {...properties}
    >
      {children}
    </DropdownMenuPrimitive.Content>
  </DropdownMenuPrimitive.Portal>
));

DropdownMenuContent.displayName = DropdownMenuPrimitive.Content.displayName;

export {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuGroup,
  DropdownMenuPortal,
  DropdownMenuTrigger,
};
