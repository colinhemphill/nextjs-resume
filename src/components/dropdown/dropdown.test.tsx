import { DropdownMenuRadioItem } from '@radix-ui/react-dropdown-menu';
import { render } from '@test-utils';
import { describe, expect, test } from 'bun:test';
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuTrigger,
} from 'src/components/dropdown/dropdown-menu';
import { DropdownMenuLabel } from 'src/components/dropdown/dropdown-menu-label';
import { DropdownMenuRadioGroup } from 'src/components/dropdown/dropdown-menu-radio';
import { DropdownMenuSeparator } from 'src/components/dropdown/dropdown-menu-separator';

describe('<Dropdown />', () => {
  test('Snapshot', () => {
    const { asFragment } = render(
      <DropdownMenu>
        <DropdownMenuTrigger>Open dropdown</DropdownMenuTrigger>
        <DropdownMenuContent>
          <DropdownMenuLabel>Menu</DropdownMenuLabel>
          <DropdownMenuSeparator />
          <DropdownMenuRadioGroup>
            <DropdownMenuRadioItem value="light">Light</DropdownMenuRadioItem>
            <DropdownMenuRadioItem value="dark">Dark</DropdownMenuRadioItem>
            <DropdownMenuRadioItem value="system">System</DropdownMenuRadioItem>
          </DropdownMenuRadioGroup>
        </DropdownMenuContent>
      </DropdownMenu>,
    );
    expect(asFragment).toMatchSnapshot();
  });
});
