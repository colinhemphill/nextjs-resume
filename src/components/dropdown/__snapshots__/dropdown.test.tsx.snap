// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Snapshot 1`] = `[Function]`;
